app/models/article.rb

```
@@ -963,15 +963,15 @@ def replace_blank_title_for_status
   def restrict_attributes_with_status_types
     # Return early if this is already saved and the body_markdown hasn't changed
     return if persisted? && !body_markdown_changed?
+    return unless type_of == "status"
+    return if body_url.present?
 
     # For status types, allow body_markdown only if it contains embed tags from URLs in title
-    if type_of == "status" && body_url.blank? && body_markdown.present?
+    if body_markdown.present?
       # Allow body_markdown if it only contains embed tags that were added from URLs in title
       unless body_markdown_only_contains_embed_tags_from_title?
```

RuboCop (1.80.2)

[convention] Use a guard clause (`return if body_markdown_only_contains_embed_tags_from_title?`) instead of wrapping the code inside a conditional expression. (https://rubystyle.guide#no-nested-conditionals)

(Style/GuardClause)